mne/epochs.py

```diff
@@ -746,13 +746,29 @@ def _reject_setup(self, reject, flat):
                    'previous ones')
         for key in set(reject.keys()).union(old_reject.keys()):
             old = old_reject.get(key, np.inf)
-            new = reject.get(key, np.inf)
+            if key in old_reject and key not in reject:
+                # The newly-passed `reject` parameter contains fewer channel
+                # types than the previous one. For the omitted channel types,
+                # simply re-use the old thresholds
+                new = old
+                reject[key] = new
+            else:
+                new = reject[key]
```

Comment:
possibly simpler / clearer?

```py
# copy prev. keys that weren't passed this time
for key in set(old_reject) - set(reject):
    reject[key] = old_reject[key]
# make sure more stringent
for key in reject:
    if reject[key] > old_reject[key]:
        raise ValueError(...)
```

Reply:
Good idea. I didn't want to touch too much of the existing implementation, but I like your proposal. I will test to see if this works.

Reply:
I've refactored the code based on your suggestion